### ISSUE TITLE
Code Editor: Ctrl-D to duplicate Code results in asymmetric behavior

### ISSUE THREAD

**Issue by iampremo** [YoYoGames/GameMaker-Bugs#1075]
GM-3904
Bug Type: Bug in the UI or IDE functionality
Ide: 2022.2.0.614
Runtime: 2022.2.0.487
Os: Windows 10
Components: Code Editor

Description: Pressing Ctrl-D without any selected code in a line duplicates that line. This is a good shortcut, thanks for that!

But when Code is selected, the behavior is wrong, the duplicate appears at the wrong position and even moves code around.

How to reproduce:
Paste these demo-lines in any code window:
//---------
firstVar = "1";
secondVar = "2";
thirdVar = "3";
fourthVar = "4"
//---------

Now do the following:
1) Put the cursor anywhere in the line of "secondVar" and press Ctrl-D
The code now looks like this:
//---------
firstVar = "1";
secondVar = "2";
secondVar = "2";
thirdVar = "3";
fourthVar = "4"
//---------
(Your cursor should be in the copied line of secondVar, at the end of the line after that)

Now press POS1/HOME key to move the cursor to the left and select two lines of code by using Shift+CursorDown twice
You should have secondVar+thirdVar lines selected.
Press Ctrl-D again.

The _expected_ behavior is, that the code now looks like this:

//---------
firstVar = "1";
secondVar = "2";
secondVar = "2";
thirdVar = "3";
secondVar = "2";
thirdVar = "3";
fourthVar = "4"
//---------

but the reality is, the code "skipped" over the fourthVar line and instead the code looks like this:
//---------
firstVar = "1";
secondVar = "2";
secondVar = "2";
thirdVar = "3";
fourthVar = "4"
secondVar = "2";
thirdVar = "3";
//---------

When selecting multiple lines, somehow, there is another line break pasted/inserted, this even results in empty lines and other unexpected results.

The same wrong behavior can be seen when using Ctrl-K to comment a block of code.
If code is not selected with the mouse, but with the keyboard using shift+cursordown (what most developers do, I think), and you press Ctrl-K to comment a selected number of lines, the LINE OF THE CURSOR is also commented, even if that line is not selected!
So I have to always press cursor-left to "deselect" that last linebreak to get the desired result.

This is a totally unexpected and unusual behavior compared to text editors of other IDEs.

Short Version:
I think the problem is the last CR/LF character counts as selected. It is taken fully into account for alle those operations.
Possible fix would be: If (selection_ends_with_crlf) remove_last_crlf_from_selection
